src/nvim/os/stdpaths.c

```diff
@@ -0,0 +1,65 @@
+#include "nvim/os/os.h"
+#include "nvim/strings.h"
+#include "nvim/path.h"
+#include "nvim/garray.h"
+
+typedef enum {
+  config_home,
+  data_home,
+  cache_home,
+  runtime_dir,
+  config_dirs,
+  data_dirs,
+} XDGDirType;
+
+static const char *xdg_env_vars[] = {
+  [config_home] = "XDG_CONFIG_HOME",
+  [data_home] = "XDG_DATA_HOME",
+  [cache_home] = "XDG_CACHE_HOME",
+  [runtime_dir] = "XDG_RUNTIME_DIR",
+  [config_dirs] = "XDG_CONFIG_DIRS",
+  [data_dirs] = "XDG_DATA_DIRS",
+};
+
+#ifdef __linux__
+static const char *const xdg_defaults[] = {
+  [config_home] = "~/.config",
+  [data_home] = "~/.local/share",
+  [cache_home] = "~/.cache",
+  [runtime_dir] = "",
+  [config_dirs] = "/etc/xdg/",
+  [data_dirs] = "/usr/local/share/:/usr/share/",
```

Comment:
Just following the spec: http://standards.freedesktop.org/basedir-spec/basedir-spec-latest.html